internal/entitlement/balanceworker/worker.go

```diff
@@ -121,30 +128,30 @@ func New(opts WorkerOptions) (*Worker, error) {
 		middleware.Recoverer,
 	)
 
-	if opts.PoisonQueue != nil {
-		poisionQueue, err := middleware.PoisonQueue(opts.Publisher, opts.PoisonQueue.Topic)
+	if opts.DLQ != nil {
+		dlq, err := wmmiddleware.DLQ(opts.Publisher, opts.DLQ.Topic, nil)
```

Comment:
We should use the poison queue implementation from Watermill and add additional middleware that does the splitting in this specific case.

I don't want to maintain a full Dlq implementation when the library provides one.

From the Schemantics perspective I would expect 3 cases from the PoisionQueue + DLQ implementation:
- If an error happens, and it is a known non-retriable error => drop from DLQ
- If an ingest split is needed, then let's only DLQ the batch items for failing entries
- Else DLQ

Reply:
I don't understand what you mean by not wanting to maintain a full DLQ implementation, as you see its roughly 50 lines split into two methods

About the expected behavior:
- the main difference between the current implementation and watermill's poisonqueu is that the current implementation only DLQs "known retryable" errors
- i see the rational behind managing it as an opt out rather than an opt in, so we could change the default to something like
```go
// defaultProcessor listens for RetryableErrors and publishes the messages contained in that
var defaultProcessor DLQMsgProcessorFunc = func(original *message.Message, err error) []*message.Message {
	if retryable, ok := errorsx.ErrorAs[watermill.RetryableError](err); ok {
		return retryable.RetryMessages()
	}

	return []*message.Message{original}
}
```
which i think gives best of both

thoughts?

Reply:
My toughts are that the PoisionQueue implementation is perfect in watermill. The only thing we need is such a handler for this specific use-case that handles the specific case of ingress batch split.